src/plugins/trace/dll/DataModel/ETW/QuicEtwEvent.cs

```diff
@@ -101,209 +103,215 @@ private static QuicObjectType ComputeObjectType(TraceEvent evt)
             }
         }
 
-        internal static unsafe QuicEvent? TryCreate(TraceEvent evt, Timestamp timestamp)
+        // TODO: QuicEtwDataReader to IDataReader
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Style", "IDE2001:Embedded statements must be on their own line", Justification = "<Pending>")]
+        internal static unsafe QuicEvent? TryCreate(Timestamp timestamp, QuicEventId id, ushort processor, uint processId, uint threadId, int pointerSize, QuicEtwDataReader data)
         {
-            var id = (QuicEventId)evt.ID;
-            var processor = (ushort)evt.ProcessorNumber;
-            var processId = (uint)evt.ProcessID;
-            var threadId = (uint)evt.ThreadID;
-            var pointerSize = evt.PointerSize;
-            var data = new QuicEtwDataReader(evt.DataStart.ToPointer(), evt.EventDataLength, pointerSize);
-
-            switch (id)
+            switch (id.ToString())
```

Comment:
This shouldn't be necessary for ETW since we do have the ID to work with. Only LTTng event processing would need to do it by name.